packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Legend/index.tsx

```diff
@@ -42,88 +42,87 @@ const LegendPanel: FC<Props> = ({ items }) => {
+            ...dirtyEchartsConfig?.legend,
+        }),
+        [dirtyEchartsConfig?.legend],
+    );
 
-    React.useEffect(() => {
-        const subscription = watch((value, { name }) => {
-            if (name && triggerSubmitFields.includes(name)) {
-                setLegend(value);
-            }
-        });
-        return () => subscription.unsubscribe();
-    }, [handleSubmit, setLegend, watch]);
+    const handleChange = (
+        prop: string,
+        newValue: string | boolean | undefined,
+    ) => {
+        const newState = { ...legendConfig, [prop]: newValue };
+        setLegend(newState);
+        return newState;
+    };
 
     const showDefault = (dirtyEchartsConfig?.series || []).length > 1;
     return (
-        <Form
-            name="legend"
-            methods={methods}
-            onSubmit={() => undefined}
-            onBlur={handleSubmit(setLegend)}
-        >
+        <Stack>
             <Checkbox
-                name="show"
-                checkboxProps={{ label: 'Show legend' }}
-                defaultValue={showDefault}
+                label="Show legend"
+                checked={legendConfig.show ?? showDefault}
+                onChange={(e) => handleChange('show', e.currentTarget.checked)}
             />
-
-            <Collapse
-                isOpen={
-                    dirtyEchartsConfig?.legend
-                        ? dirtyEchartsConfig?.legend.show
-                        : showDefault
-                }
-            >
-                <SectionTitle>Scroll</SectionTitle>
+            <Collapse in={legendConfig.show ?? showDefault}>
                 <Switch
-                    large
+                    label="Scroll legend"
+                    mb="md"
                     checked={dirtyEchartsConfig?.legend?.type !== 'plain'}
-                    onChange={() => {
-                        const type: 'scroll' | 'plain' =
-                            dirtyEchartsConfig?.legend?.type !== 'plain'
-                                ? 'plain'
-                                : 'scroll';
-                        methods.setValue('type', type);
-                        handleSubmit(setLegend)();
-                    }}
+                    onChange={(e) =>
+                        handleChange(
+                            'type',
+                            e.currentTarget.checked ? 'scroll' : 'plain',
+                        )
+                    }
                 />
-                <SectionTitle>Position</SectionTitle>
-
+                <Text fw={600}>Position</Text>
                 {[
                     [Positions.Left, Positions.Right],
                     [Positions.Top, Positions.Bottom],
                 ].map((positionGroup) => (
-                    <SectionRow key={positionGroup.join(',')}>
+                    <SimpleGrid
+                        cols={2}
+                        mt="xs"
+                        ml="xs"
+                        mb="md"
```

Comment:
I feel like at least `ml` is not necessary here
![CleanShot 2023-07-27 at 18 09 47@2x](https://github.com/lightdash/lightdash/assets/962095/5f161b8d-fa16-4512-bc86-3ebb36616f1f)

Reply:
sorry, I thought it was scrolling legend.

Reply:
so at least labels are aligned to scan text easily

Reply:
1. Good point about the Show legend/Scroll legend switches. It is a little weird that one is a switch and one is a check box. Should I just make them both one or the other? I changed the name of "Scroll" to "Scroll legend" because I though it was clearer, but if you think it would make more sense I could change it back. Or the scroll one could get moved. They have a little different scope -- one decides if you want a legend at all and the other says what kind -- giving them the same visual weight right next to each other is a little odd. But maybe I'm overthinking it 😄.

2. I put the left margin there on purpose cause I think it makes it more clear that all of those boxes are sub-elements of Position. It looked a little funny without it. I still think it looks clearer, but if you don't agree, no problem.

Reply:
I like the spacing to make the sub-elements more obvious and I agree that switches are the best way :)